components/fxa-client/src/oauth.rs

```diff
@@ -82,6 +82,27 @@ impl FirefoxAccount {
         Ok(token_info)
     }
 
+    /// Check whether access token is valid using the saved refresh token.
+    /// If there is no refresh token held or if it is not authorized,
+    /// Error is thrown.
+    pub fn check_authorization_status(&mut self) -> Result<IntrospectInfo> {
```

Comment:
It would be useful to add some basic test coverage for this, but since many of the existing methods in this file don't have great test coverage, I won't block on it for this one either. If you'd like to give it a shot, you can see an example of how to mock out the network requests here:

   https://github.com/mozilla/application-services/blob/master/components/fxa-client/src/profile.rs#L115